mxcube3/routes/Login.py

```diff
@@ -42,21 +44,36 @@ def login():
 
     try:
         login_res = limsutils.lims_login(loginID, password)
-        inhouse = limsutils.lims_is_inhouse(login_res)
 
+        if login_res['status']["code"] is not "ok":
+            return deny_access("Could not authenticate")
+
+        inhouse = limsutils.lims_is_inhouse(login_res)
         info = {"valid": limsutils.lims_valid_login(login_res),
                 "local": is_local_host(),
                 "existing_session": limsutils.lims_existing_session(login_res),
                 "inhouse": inhouse}
 
-        _users = logged_in_users(exclude_inhouse=True)
+        _users = logged_in_users(exclude_inhouse=False)
+
+        common_proposal = False
+        if mxcube.SELECTED_PROPOSAL is not None:
```

Comment:
So we dont have to have a selected proposal to login ?

Reply:
When a first user authenticates, then he will select a proposal (via SelectProposal form, call to /lims/proposal) or it will be autoselected if `loginType != user` (call to logintutils.set_operator -> select_proposal ).

At that point mxcube.SELECTED_PROPOSAL will have the proposal name.

When another user tries to login, mxcube checks first if the login is valid and then if there is a proposal already selected. This `if` clause is to determine if they share a proposal, for later in 
```
        if not is_staff and not common_proposal and _users and (loginID not in _users):
            return deny_access("Another user is already logged in")
``` 

Which is probably not the best error message though...

We are using this logic in prod here with remote, multiple users, staff etc. But I could not tested very deeply with your login method. Anyway, maintaining both login methods is not easy so let me know if you have ideas to make things clearer.